core/src/Cabana_SoA.hpp

```diff
@@ -337,20 +336,27 @@ struct SoA<MemberTypes<Types...>,VectorLength>
 
     // Get a pointer to a member.
     template<std::size_t M>
-    void* ptr()
+    KOKKOS_FUNCTION void* ptr()
     {
         Impl::StructMember<M,vector_length,member_data_type<M> >& base = *this;
         return &base;
     }
+
+    // Get a pointer to a given SoA.
+    template<std::size_t M>
+    static void* staticPtr( SoA* p )
+    {
+        Impl::StructMember<M,vector_length,member_data_type<M> >& base = *p;
+        return &base;
```

Comment:
Would you explain how `soa_type::template staticPtr<M>(ptr)` differs from `static_cast<void*>(ptr)`?
What am I missing?

Reply:
I missed the multiple inheritance of `SoAImpl` from `SructMember<0, T0>`,  `SructMember<1, T1>`, etc.
This is an implicit conversion to a sub-object of the struct.

I think we could just do `return static_cast<Impl::StructMember<M,vector_length,member_data_type<M>>*>(p);`

Reply:
added a commit for this